### PR TITLE
:lipstick: Clean up core classes

### PR DESCRIPTION
**Pull request by tomrijnbeek**

### CODE REVIEW

File: src/amulware.Graphics/Core/Color.cs [hunk 1]
```diff
@@ -363,42 +363,42 @@ public Color(Color color, byte newAlpha)
         /// <returns>The constructed colour.</returns>
         public static Color FromHSVA(float h, float s, float v, byte a = 255)
         {
-            float chroma = v * s;
+            var chroma = v * s;
             h /= MathHelper.PiOver3;
-            float x = chroma * (1 - Math.Abs((h % 2) - 1));
-            float m = v - chroma;
+            var x = chroma * (1 - Math.Abs(h % 2 - 1));
```
Comment (amulware) [inline src/amulware.Graphics/Core/Color.cs]:
```suggestion
            var x = chroma * (1 - Math.Abs((h % 2) - 1));
```
This is not entirely obvious to me.

File: src/amulware.Graphics/Core/Color.cs [hunk 2]
```diff
@@ -363,42 +363,42 @@ public Color(Color color, byte newAlpha)
         /// <returns>The constructed colour.</returns>
         public static Color FromHSVA(float h, float s, float v, byte a = 255)
```
Comment (amulware) [inline src/amulware.Graphics/Core/Color.cs]: Should we spell out the parameter names here?

File: src/amulware.Graphics/Core/Color.cs [hunk 3]
```diff
@@ -524,25 +521,28 @@ public float Hue
...
-                float b = this.b / 255f;
+                var floatR = r / 255f;
+                var floatG = g / 255f;
+                var floatB = b / 255f;
 
                 float h;
 
-                var max = Math.Max(r, Math.Max(g, b));
+                var max = Math.Max(floatR, Math.Max(floatG, floatB));
+                // ReSharper disable once CompareOfFloatsByEqualityOperator
```
Comment (amulware) [inline src/amulware.Graphics/Core/Color.cs]: Disable for the entire block and enable again at the end?

File: src/amulware.Graphics/Core/Color.cs [hunk 4]
```diff
@@ -560,54 +560,46 @@ public Vector4 AsHSVAVector
...
-                float b = this.b / 255f;
-                float a = this.a / 255f;
+                var floatR = r / 255f;
+                var floatG = g / 255f;
+                var floatB = b / 255f;
+                var floatA = a / 255f;
 
-                var max = Math.Max(r, Math.Max(g, b));
+                var max = Math.Max(floatR, Math.Max(floatG, floatB));
+                // ReSharper disable once CompareOfFloatsByEqualityOperator
```
Comment (amulware) [inline src/amulware.Graphics/Core/Color.cs]: Disable for the whole block?

File: src/amulware.Graphics/Core/Color.cs [hunk 5]
```diff
@@ -635,40 +626,32 @@ public Color Premultiplied
...
-            return this.r == other.r && this.g == other.g && this.b == other.b && this.a == other.a;
-        }
+        public bool Equals(Color other) => r == other.r && g == other.g && b == other.b && a == other.a;
 
         /// <summary>
         /// Checks whether this colour is the same as another.
         /// </summary>
         public override bool Equals(object obj)
         {
             if (ReferenceEquals(null, obj)) return false;
```
Comment (amulware) [inline src/amulware.Graphics/Core/Color.cs]: Remove this line. It's not needed with the `is Color` check.

File: src/amulware.Graphics/Core/Color.cs (hunk 2 above)
Comment (tomrijnbeek) [reply src/amulware.Graphics/Core/Color.cs]: Done. Then again, not sure how far we want to go. Should we spell out `r, g, b, a` variables as well?

File: src/amulware.Graphics/Core/Color.cs (hunk 3 above)
Comment (tomrijnbeek) [reply src/amulware.Graphics/Core/Color.cs]: I can see why that is nicer. I'll change it, but not without pointing out my reasoning for doing it this way: I still think the warning is valuable, and if there is ever new float logic added within the block that _shouldn't_ compare floats by equality, we'll lose the warning by disabling it for the whole block. It's a trade-off, but in this case I think it's fine.

File: src/amulware.Graphics/Core/Color.cs (hunk 2 above)
Comment (amulware) [reply src/amulware.Graphics/Core/Color.cs]: In public interfaces perhaps? Then again, rgb is very obvious to me. everything else potentially less so.

File: src/amulware.Graphics/Core/Color.cs (hunk 3 above)
Comment (amulware) [reply src/amulware.Graphics/Core/Color.cs]: I fully agree with your thinking, but I think in this case it's nicer. It's very unlikely this code will be changed significantly.

**Pull request merged**